app/addons/dataimporter/stores.js

```diff
@@ -0,0 +1,369 @@
+      return this._rowsShown;
+    },
+
+    getTheMetadata: function () {
+      return this._theMetadata;
+    },
+
+    getTheData: function () {
+      return this._theData;
+    },
+
+    papaparse: function () {
+      //for some reason this._config.complete gets overwritten on setconfig
+      this._config.complete = this._completeFn;
+      this._config.error = this._errorFn;
+      Papa.parse(this._theFile, this._config);
+    },
+
+    loadingComplete: function (results) {
+      this.loadMeta(results.meta);
+      this.loadData(results.data);
+      this._totalRows = this._theData.length;
+
+      if (this.isThisABigFile()) {
+        this.calcSmallPreviewOfData();
+      }
+
+      this.chunkData();
+      this.dataLoaded();
+      this.triggerChange();
+    },
+
+    chunkData: function () {
+      for (var i = 0; i < this._theData.length; i += 500) {
+        var oneChunk = this._theData.slice(i, i + 500);
+        this._chunkedData.push(oneChunk);
+      }
+    },
+
+    clearData: function () {
+      this._theData = [];
+    },
+
+    setParseConfig: function (key, value) {
+      this._config[key] = value;
+    },
+
+    getConfigSetting: function (key) {
+      return this._config[key];
+    },
+
+    getDefaultConfig: function () {
+      return {
+        delimiter : '',  // auto-detect
+        newline: '',  // auto-detect
+        header: true,
+        dynamicTyping: true,
+        preview: 0,
+        encoding: '',
+        worker: true, //true = page doesn't lock up
+        comments: false,
+        complete: function (results) {
+          this.loadingComplete(results);
+          this._changingConfig = false;
+        }.bind(this),
+        error: function () {
+          var msg1 = 'There was an error while parsing the file.',
+              msg2 = 'Please try again.';
+
+          this.goToErrorScreen('', [msg1, msg2]);
+        }.bind(this),
+        download: false,
+        skipEmptyLines: false,
+        chunk: undefined, //define function for streaming
+        beforeFirstChunk: undefined,
+      };
+    },
+
+    loadDataIntoDatabase: function (createNewDB, targetDB) {
+      this._loadingInDBInProgress = true;
+      this.triggerChange();
+      if (createNewDB) {
+        if (this.dataBaseIsnew(targetDB)) {
+          var msg1 = 'The database ' + targetDB + ' already exists.';
+          this.goToErrorScreen('', [msg1]);
+        }
+        this.loadIntoNewDB(targetDB);
+      } else {
+        this.loadDataIntoTarget(targetDB);
+      }
+    },
+
+    dataBaseIsnew: function (targetDB) {
+      return _.some(this._all_dbs, targetDB);
+    },
+
+    showErrorScreen: function () {
+      return this._showErrorScreen;
+    },
+
+    getErrorMsg: function () {
+      return this._errorMessageArray;
+    },
+
+    goToErrorScreen: function (resp, messageArray) {
+      this._loadingInDBInProgress = false;
+      this._showErrorScreen = true;
+      if (resp) messageArray.push(resp);
```

Comment:
I'm surprised this doesn't freak out the style checker. All `if`'s should have curly braces, even one liners.